config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/Encass.java

```diff
@@ -117,6 +125,28 @@ public void setPath(String path) {
         this.path = path;
     }
 
+    public boolean isParentEntityShared() {
+        return isParentEntityShared;
+    }
+
+    public void setParentEntityShared(boolean parentEntityShared) {
+        isParentEntityShared = parentEntityShared;
+    }
+
+    Encass merge(Encass otherEncass) {
```

Comment:
I'm not sure about the correctness of this logic. I don't think some of the attributes (ex: guid, name, etc) cannot be derived from others. I might be early to comment on this.

Reply:
Understood. You are cloning the encass object. Keeping it as separate method gives different meaning. I recommend you to move inline to the copy-constructor.

Reply:
Moved to the constructor.